cli/tools/lint.rs

```diff
@@ -373,3 +415,57 @@ fn sort_diagnostics(diagnostics: &mut Vec<LintDiagnostic>) {
     }
   });
 }
+
+fn get_rules_from_config(
+  maybe_lint_config: Option<&LintConfig>,
+) -> Result<Vec<Box<dyn LintRule>>, AnyError> {
+  let configured_rules = if let Some(lint_config) = maybe_lint_config {
+    let mut filtered_rules = HashMap::new();
+
+    if let Some(tags) = &lint_config.rules.tags {
+      for config_tag in tags {
+        filtered_rules.extend(rules::get_all_rules().into_iter().filter_map(
+          |rule| {
+            let code = rule.code();
+            if rule.tags().contains(&config_tag.as_str()) {
+              Some((code, rule))
+            } else {
+              None
+            }
+          },
+        ));
+      }
+    }
+
+    if let Some(exclude) = &lint_config.rules.exclude {
+      for exc in exclude {
+        filtered_rules.remove(exc.as_str());
+      }
+    }
+
+    if let Some(include) = &lint_config.rules.include {
+      let mut rules_per_code = rules::get_all_rules()
+        .into_iter()
+        .map(|rule| (rule.code(), rule))
+        .collect::<HashMap<_, _>>();
+      for inc in include {
+        if let Some(rule) = rules_per_code.remove(inc.as_str()) {
+          filtered_rules.insert(inc, rule);
+        }
+      }
+    }
+
+    filtered_rules
+      .into_iter()
+      .map(|(_code, rule)| rule)
+      .collect()
+  } else {
+    rules::get_recommended_rules()
+  };
+
+  if configured_rules.is_empty() {
+    anyhow!("No rules have been configured");
```

Comment:
Maybe we can suggest to add `"tags": ["recommended"]` if there's no `tags` property? That will probably address my scenario above.

Reply:
"recommended" is enabled by default - you get this error only if you do `"tags": []` and don't specify anything in `rules.include`